src/tests/test_bootstrap.py

```diff
@@ -25,12 +25,34 @@ def test_valid_master(self, popen, mocked_send_usage):
             self.assertTrue(mocked_send_usage.called)
 
     @mock.patch('subprocess.Popen')
-    def test_valid_slave(self, mocked_popen):
+    def test_master_not_ready_in_time(self, popen):
+        os.environ['ROLE'] = 'slave'
+        os.environ['TARGET_HOST'] = 'https://test.com'
+        os.environ['MASTER_HOST'] = '127.0.0.1'
+        os.environ['SLAVE_MUL'] = '1'
+        os.environ['MASTER_CHECK_TIMEOUT'] = '0.3'
+        os.environ['MASTER_CHECK_INTERVAL'] = '0.1'
```

Comment:
Rather than modifying the real environment (which is global state) and forgetting to clean it up afterwards, I would rather suggest using [`monkeypatch.setenv`](https://docs.pytest.org/en/latest/reference.html#_pytest.monkeypatch.MonkeyPatch.setenv) since it looks like you're using Pytest:

```py
@mock.patch('subprocess.Popen')
def test_master_not_ready_in_time(self, monkeypatch, popen):  # not sure of the ordering between monkeypatch & popen
  monkeypatch.setenv('ROLE', 'slave')
  ...
```

Reply:
If it's applicable, I'd rather make this modification for all unit tests for consistency and as a part of the other ticket.

Reply:
Sure, but could we create that other ticket now? I'm pretty sure I would forget otherwise.

Reply:
https://github.bus.zalan.do/TIP/docs/issues/347